api/types/src/bytecode.rs

```diff
@@ -131,8 +135,9 @@ pub trait Bytecode {
         }
     }
 
-    fn new_move_struct(&self, def: &StructDefinition) -> MoveStruct {
+    fn new_move_struct(&self, def: &StructDefinition, is_option_module: bool) -> MoveStruct {
```

Comment:
No need to eagerly compute `is_option_module`. You can pass in `mod_of_def: &ModuleId`.